test/mlir/onnx/onnx_bundle_memory_pool.mlir

```diff
@@ -1,4 +1,4 @@
-// RUN: onnx-mlir-opt --shape-inference --convert-onnx-to-krnl --buffer-deallocation --enable-memory-pool --bundle-memory-pools --canonicalize %s -split-input-file | FileCheck %s
+// RUN: onnx-mlir-opt --shape-inference --convert-onnx-to-krnl='opt-level=3' --buffer-deallocation --enable-memory-pool --bundle-memory-pools --canonicalize %s -split-input-file | FileCheck %s
```

Comment:
The general problem here is that certain passes can lower the code in a naive way, or lower the code in an optimized way. That is a pass can have some basic functionality and a set of additional options to control the behaviour of the pass. 

The same problem arises within LLVM passes, as an example consider `LoopUnroll` (llvm/lib/Transforms/Scalar/LoopUnrollPass.cpp). That pass has itself a set of options to control its behaviour: 

```
static cl::opt<unsigned>
    UnrollThreshold("unroll-threshold", cl::Hidden,
                    cl::desc("The cost threshold for loop unrolling"));

static cl::opt<unsigned>
    UnrollOptSizeThreshold(
      "unroll-optsize-threshold", cl::init(0), cl::Hidden,
      cl::desc("The cost threshold for loop unrolling when optimizing for "
               "size"));
               
static cl::opt<unsigned> UnrollCount(
    "unroll-count", cl::Hidden,
    cl::desc("Use this unroll count for all loops including those with "
             "unroll_count pragma values, for testing purposes"));
... many more ... 

```

These various options can be exercised in LIT tests to get complete test coverage:

```
ARM/instr-size-costs.ll:; RUN: opt -loop-unroll -unroll-allow-partial -unroll-optsize-threshold=18 -mtriple=thumbv8 -S %s -o - | FileCheck %s --check-prefix=CHECK-V8

unroll-count.ll:; RUN: opt < %s -S -loop-unroll -unroll-count=2 | FileCheck %s
...
```

The optlevel  `onnx-mlir-op -O3  --shape-inference --convert-onnx-to-krnl . . .`  could be passed by the driver to the constructor of the various passes. The constructor would then flip the default setting of the various flags controlling pass behaviour, so with -O3 it would set the internal flags to aggressive settings. In this case it would enable SIMDization.  What do you think?

Reply:
Thanks for the suggestion, this is now how things are done.